lib/omniauth/strategies/openid_connect.rb

```diff
@@ -143,10 +158,17 @@ def issuer
       end
 
       def discover!
-        client_options.authorization_endpoint = config.authorization_endpoint
-        client_options.token_endpoint = config.token_endpoint
-        client_options.userinfo_endpoint = config.userinfo_endpoint
-        client_options.jwks_uri = config.jwks_uri
+        return unless options.discovery
+        options.issuer = issuer if options.issuer.blank?
+        setup_client_options(config)
```

Comment:
1. what's the point to pass `config` as argument?
2. what's the point of extraction `setup_client_options` method? how does it pay for itself?
3. please avoid using Rails methods in gems which can be used outside of Rails project (`try`)

Consider the following changes:

```ruby
def discover!
  return unless options.discovery
  client_options.authorization_endpoint = config.authorization_endpoint
  client_options.token_endpoint = config.token_endpoint
  client_options.userinfo_endpoint = config.userinfo_endpoint
  client_options.jwks_uri = config.jwks_uri
  client_options.end_session_endpoint = config.end_session_endpoint if config.respond_to?(:end_session_endpoint)
end
```

Reply:
Totally agree, I can't explain why I did that.
Thanks for your feedback.